pkg/geo/geomfn/add_measure_test.go

```diff
@@ -0,0 +1,193 @@
+			input:    geom.NewLineString(geom.XYZ),
+			start:    0,
+			end:      1,
+			expected: geom.NewLineString(geom.XYZM),
+		},
+		{
+			desc:     "add measure to empty 4D linestring",
+			input:    geom.NewLineString(geom.XYZM),
+			start:    0,
+			end:      1,
+			expected: geom.NewLineString(geom.XYZM),
+		},
+		{
+			desc:     "add measure to 2D multilinestring",
+			input:    geom.NewMultiLineStringFlat(geom.XY, []float64{0, 0, 1, 1}, []int{4}),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineStringFlat(geom.XYM, []float64{0, 0, 0, 1, 1, 1}, []int{6}),
+		},
+		{
+			desc:     "add measure to 2D multilinestring with empty",
+			input:    geom.NewMultiLineStringFlat(geom.XY, []float64{0, 0, 1, 1}, []int{4, 4}),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineStringFlat(geom.XYM, []float64{0, 0, 0, 1, 1, 1}, []int{6, 6}),
+		},
+		{
+			desc:     "add measure to 2D+M multilinestring",
+			input:    geom.NewMultiLineStringFlat(geom.XYM, []float64{0, 0, -5, 1, 1, 5}, []int{6}),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineStringFlat(geom.XYM, []float64{0, 0, 0, 1, 1, 1}, []int{6}),
+		},
+		{
+			desc:     "add measure to 3D multilinestring",
+			input:    geom.NewMultiLineStringFlat(geom.XYZ, []float64{0, 0, -5, 1, 1, 5}, []int{6}),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineStringFlat(geom.XYZM, []float64{0, 0, -5, 0, 1, 1, 5, 1}, []int{8}),
+		},
+		{
+			desc:     "add measure to 4D multilinestring",
+			input:    geom.NewMultiLineStringFlat(geom.XYZM, []float64{0, 0, -5, 23, 1, 1, 5, -23}, []int{8}),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineStringFlat(geom.XYZM, []float64{0, 0, -5, 0, 1, 1, 5, 1}, []int{8}),
+		},
+		{
+			desc:     "add measure to empty 2D multilinestring",
+			input:    geom.NewMultiLineString(geom.XY),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineString(geom.XYM),
+		},
+		{
+			desc:     "add measure to empty 2D+M multilinestring",
+			input:    geom.NewMultiLineString(geom.XYM),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineString(geom.XYM),
+		},
+		{
+			desc:     "add measure to empty 3D multilinestring",
+			input:    geom.NewMultiLineString(geom.XYZ),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineString(geom.XYZM),
+		},
+		{
+			desc:     "add measure to empty 4D multilinestring",
+			input:    geom.NewMultiLineString(geom.XYZM),
+			start:    0,
+			end:      1,
+			expected: geom.NewMultiLineString(geom.XYZM),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			geometry, err := geo.MakeGeometryFromGeomT(tc.input)
+			require.NoError(t, err)
+
+			got, err := AddMeasure(geometry, tc.start, tc.end)
+			require.NoError(t, err)
+
+			want, err := geo.MakeGeometryFromGeomT(tc.expected)
+			require.NoError(t, err)
+
+			require.Equal(t, want, got)
+			require.EqualValues(t, tc.input.SRID(), got.SRID())
+		})
+	}
```

Comment:
can you add the error cases (!linestring !multilinestring) as a separate case below here? can be a separate `errorTestCases` loop